### PR TITLE
Brand Loading

### PR DESCRIPTION
**Pull request by gvonkoss**

### CODE REVIEW

File: src/scss/_variables.scss [hunk 1]
```diff
@@ -16,11 +16,8 @@ $o-loading-animation-keyframes: false !default;
 
 /// Predefined color themes
 ///
-/// @type Map
-$o-loading-colors: (
-	'light': oColorsGetPaletteColor('white'),
-	'dark': oColorsGetPaletteColor('black-80')
-);
+/// @type List
+$o-loading-colors: ('light', 'dark');
```
Comment (JakeChampion) [inline src/scss/_variables.scss]: Does changing the data type from Map to List make this a breaking change?

Comment (gvonkoss) [reply src/scss/_variables.scss]: Indeed you are technically correct. Making a change now.

File: demos/src/demo.scss [hunk 2]
```diff
@@ -13,11 +17,19 @@ $o-loading-is-silent: false;
 }
 
 .demo-card--light {
-	background: #f6e9d8;
+	@if (global-variable-exists(o-brand)) {
+		background: if($o-brand == 'internal', white, #fff1e5);
+	} @else {
```
Comment (notlee) [inline demos/src/demo.scss]: Did `@include oColorsFor(page);` not work out?

File: demos/src/demo.scss [hunk 3]
```diff
@@ -13,11 +17,19 @@ $o-loading-is-silent: false;
...
+		background: #fff1e5;
+	}
 }
 
 .demo-card--dark {
-	background: #333333;
+	@if (global-variable-exists(o-brand)) {
+		background: if($o-brand == 'internal', #262a33, #333333);
```
Comment (notlee) [inline demos/src/demo.scss]: In other demos lots of our dark/inverse backgrounds for master are slate. Can't this be slate for all cases?

File: demos/src/demo.scss (hunk 2 above)
Comment (gvonkoss) [reply demos/src/demo.scss]: No it didn't. `@include oColorsGetUseCase(page, background)` didn't either. I don't know if it had something to do with the way my demos kept building, but even after removing and reinstalling everything, I got nothing. If you manage to make it work locally, I'll change it.

File: demos/src/demo.scss (hunk 3 above)
Comment (gvonkoss) [reply demos/src/demo.scss]: That's just was it was, I don't really mind either way.

File: src/scss/_brand.scss [hunk 4]
```diff
@@ -0,0 +1,39 @@
+
+/// Output a color for a variant of a brand
+///
+/// @private
+
+// Lives inside brand file because of source order.
+@function _oLoadingGetColors($variant) {
+	@return oBrandGet($component: 'o-loading', $variables: 'default-color', $configure-for: $variant);
```
Comment (notlee) [inline src/scss/_brand.scss]: I don't _think_ this is needed, can the `oBrandGet` call be in `src/scss/_variables.scss`?

File: src/scss/_mixins.scss [hunk 5]
```diff
@@ -58,7 +58,7 @@
 /// @param {String|Number} $size
 @mixin oLoadingSize($size) {
 	@if type-of($size) == 'string' {
-		$size: map-get($o-loading-sizes, $size);
+		$size: indexof($o-loading-sizes, $size);
```
Comment (notlee) [inline src/scss/_mixins.scss]: Shouldn't this stay as `map-get`? My interpretation is that if `$size` is a number it represents a pixel value and shouldn't get from the `$o-loading-sizes` map.

File: demos/src/demo.scss (hunk 2 above)
Comment (notlee) [reply demos/src/demo.scss]: Seems to work! ¯\_(ツ)_/¯

**Review by notlee — APPROVED** [review]

**Pull request merged**